rocky/rocky/views/mixins.py

```diff
@@ -55,14 +55,18 @@ class OOIAttributeError(AttributeError):
 
 
 class ObservedAtMixin:
-    def get_observed_at(self) -> datetime:
+    @cached_property
+    def observed_at(self) -> datetime:
         observed_at = self.request.GET.get("observed_at", None)
         if not observed_at:
             return datetime.now(timezone.utc)
 
         try:
             datetime_format = "%Y-%m-%d"
-            return convert_date_to_datetime(datetime.strptime(observed_at, datetime_format))
+            date_time = convert_date_to_datetime(datetime.strptime(observed_at, datetime_format))
+            if date_time.date() > datetime.now(timezone.utc).date():
+                messages.warning(self.request, _("Beware: The selected date is in the future."))
```

Comment:
It already is a warning and that already show "warning" in the user interface, so I don't we need to add `beware:` to the warning text.